lib/distrobaker.py

```diff
@@ -766,16 +794,25 @@ def get_buildsys(which):
     if 'main' not in c:
         logger.critical('DistroBaker is not configured, aborting.')
         return None
-    if which != 'source' and which != 'destination':
+    if which not in ('source', 'destination'):
         logger.error('Cannot get "%s" build system.', which)
         return None
-    if not hasattr(get_buildsys, which):
+
+    session_timed_out = False
+    if hasattr(get_buildsys, which):
+        session_age = datetime.datetime.now() - getattr(get_buildsys, which)['session_start_time']
```

Comment:
We will get `session_age` only if `get_buildsys()` method will be called and `hasattr(get_buildsys, which)` condition.
I thought we need to renew a session once per hour without additional conditions. Am I right?

Reply:
The condition is only there to check whether we're calling the function for the first time. IOW, it's always true, except on the first run, in which case the session has no age because no session exists yet. The `session_age` only gets computed on the second and every subsequent run, after `session_start_time` has been set.